File: MobileApp/AlexaDeviceFinder/AlexaDeviceFinder.Android/Utilities/AmazonLoginHelper.cs
using Android.Content;
using Android.Widget;

namespace DeviceFinder.Droid.Utilities
{
    public class AmazonLoginHelper
    {
        private Context _context;

        public static void SetUserId(Context context)
        {
            //var userFetchResult = User.Fetch(context, new Listener<User, AuthError>();
        }

        public void OnUserFetchSuccess()// User user)
        {
            PreferencesManager preferencesManager = new PreferencesManager(_context);
            preferencesManager.SetAmazonUserId("user.GetUserId()");
        }

        public void OnUserFetchError()//AuthError ae)
        {
            Toast.MakeText(_context, "Error retrieving profile information.", ToastLength.Short).Show();
        }

        public static void SignOut(Context context)
        {
            //var signOutResult = AuthorizationManager.SignOut(context, new Listener<Void, AuthError>();
        }

        public void OnSignOutSuccess()//Void response)
        {
            //Toast.MakeText(context, "Signed out", ToastLength.Short).Show();
        }

        public void OnSignOutError()//AuthError authError)
        {
            Toast.MakeText(_context, "Failed to sign out", ToastLength.Short).Show();
        }
    }
}

File: AndroidApp/app/src/main/java/com/jamespfluger/devicefinder/utilities/AmazonLoginHelper.java
package com.jamespfluger.devicefinder.utilities;

import android.content.Context;
import android.os.Handler;
import android.os.Looper;
import android.widget.Toast;

import com.amazon.identity.auth.device.AuthError;
import com.amazon.identity.auth.device.api.Listener;
import com.amazon.identity.auth.device.api.authorization.AuthorizationManager;
import com.amazon.identity.auth.device.api.authorization.User;
import com.jamespfluger.devicefinder.R;
import com.jamespfluger.devicefinder.settings.ConfigManager;

public class AmazonLoginHelper {
    public static void setUserId(final Context context) {
        User.fetch(context, new Listener<User, AuthError>() {
            @Override
            public void onSuccess(User user) {
                ConfigManager.setLoginUserId(user.getUserId());
            }

            @Override
            public void onError(AuthError ae) {
                new Handler(Looper.getMainLooper()).post(() -> Toast.makeText(context, R.string.amazon_user_fetch_error_toast, Toast.LENGTH_SHORT).show());
            }
        });
    }

    public static void signOut(final Context context) {
        AuthorizationManager.signOut(context, new Listener<Void, AuthError>() {
            @Override
            public void onSuccess(Void response) {
                new Handler(Looper.getMainLooper()).post(() -> Toast.makeText(context, R.string.amazon_sign_out_success_toast, Toast.LENGTH_SHORT).show());
            }

            @Override
            public void onError(AuthError authError) {
                new Handler(Looper.getMainLooper()).post(() -> Toast.makeText(context, R.string.amazon_sign_out_failure, Toast.LENGTH_SHORT).show());
            }
        });
    }
}
